app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt

```diff
@@ -0,0 +1,26 @@
+package net.squanchy.favorites.view
+
+private const val TAPS_TO_TRIGGER_INITIAL_ACHIEVEMENT = 5
+private const val TAPS_TO_TRIGGER_PERSEVERANCE_ACHIEVEMENT = 15
+
+private typealias AchievementMessageProvider = () -> String
+private typealias ResIdProvider = () -> Int
+
+internal fun presentButtonIcon(counter: Int, view: FavoritesSignedInEmptyLayoutView, filledIconId: ResIdProvider, emptyIconId: ResIdProvider) {
```

Comment:
I'm not really sure about this. I'm not a fan of having any Android dependency at all in a presenter. Let's wait for a feedback from @rock3r on this one

Reply:
There is no Android dependency dependency here at all. The view is just a Java interface.

Reply:
Oh I see, I confused it with `FavoritesSignedInEmptyLayout`